Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean

```diff
@@ -616,37 +610,55 @@ lemma toClass_eq_zero (P : W.Point) : toClass P = 0 ↔ P = 0 := by
 
 lemma toClass_injective : Function.Injective <| @toClass _ _ W := by
   rintro (_ | h) _ hP
-  all_goals rw [← neg_add_eq_zero, ← toClass_eq_zero, map_add, ← hP]
+  all_goals rw [← neg_inj, ← add_eq_zero, ← toClass_eq_zero, map_add, ← hP]
   · exact zero_add 0
   · exact CoordinateRing.mk_XYIdeal'_mul_mk_XYIdeal'_of_Yeq h
 #align weierstrass_curve.point.to_class_injective WeierstrassCurve.Affine.Point.toClass_injective
 
-lemma add_comm (P Q : W.Point) : P + Q = Q + P :=
-  toClass_injective <| by simp only [map_add, _root_.add_comm]
-#align weierstrass_curve.point.add_comm WeierstrassCurve.Affine.Point.add_comm
-
-lemma add_assoc (P Q R : W.Point) : P + Q + R = P + (Q + R) :=
-  toClass_injective <| by simp only [map_add, _root_.add_assoc]
-#align weierstrass_curve.point.add_assoc WeierstrassCurve.Affine.Point.add_assoc
-
-noncomputable instance instAddCommGroupPoint : AddCommGroup W.Point where
+noncomputable instance : AddCommGroup W.Point where
```

Comment:
Thanks! I have some golfs [here](https://github.com/leanprover-community/mathlib4/commit/09ad43d968d35eb6069d426d3c6e882ba31702cb).
Using [Function.Injective.addCommGroup](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/InjSurj.html#Function.Injective.addCommGroup) might be able to save us some lines but you'd still need to introduce nsmul, zsmul, and sub with three `letI`s.

Reply:
Thanks! I would hope that a friendlier version of [Function.Injective.addCommGroup](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/InjSurj.html#Function.Injective.addCommGroup) exists. Surely for any reasonable structure we can deduce `nsmul`, `zsmul`, and `sub` automatically from the others...?

Reply:
But you may not want to use the affine operations in this case. toAffine need to check whether the z coordinate is zero, which is not decidable. 

In Function.Injective.addCommGroup the SMul and Sub instances are supplied to avoid potential diamonds, which is probably not a concern here.